### PR TITLE
Update editing-escrow-contract.md

### PR DESCRIPTION
**Pull request by okwme**
Mostly trivial updates. Added comments that should be removed after being addressed. It might be good to move the reducing build size into an advanced section?

### CODE REVIEW

File: docs/intro/editing-escrow-contract.md [hunk 1]
```diff
@@ -87,7 +87,9 @@ pub fn init<T: Storage>(store: &mut T, params: Params, msg: Vec<u8>) -> Result<R
 
 You may wonder about the `clone()` in `source: params.message.signer.clone()`. This has to do with rust lifetimes. If I pass in a variable, I give "ownership" to the other structure and may no longer use it in my code. Since I need to access a reference to params later to check expiration, `state.is_expired(&params)`, I must first clone the struct. If I did not reference `params` anywhere below, I would not need the `clone`. 
 
-Try to remove the `.clone()` and compile. See what your IDE or compile says.
+Try to remove the `.clone()` and compile. See what your IDE or compile says. [__Installed rls on vscode and i get no notifications from removing the clone. Do you think i need to configure the extension somehow?__]
```
Comment (okwme) [inline docs/intro/editing-escrow-contract.md]:
```suggestion
Try to remove the `.clone()` and compile. See what your IDE or compile says.
```

File: docs/intro/editing-escrow-contract.md [hunk 2]
```diff
@@ -89,6 +89,8 @@ You may wonder about the `clone()` in `source: params.message.signer.clone()`. T
 
 Try to remove the `.clone()` and compile. See what your IDE or compile says.
 
+[__Update: VSCode didn't like being opened to the root dir of the repo. It couldn't find the Cargo.toml file in the root (or any parent dir) so couldn't run any rust logic. I turned on a multi project setup which is experimental and got some errors showing. Everything seemed to click when I closed VSCode and re-opened just the escrow directory.__]
```
Comment (okwme) [inline docs/intro/editing-escrow-contract.md]:
```suggestion
```

File: docs/intro/editing-escrow-contract.md [hunk 3]
```diff
@@ -144,14 +145,14 @@ fn try_approve(params: Params, state: State, quantity: Option<Vec<Coin>>) -> Res
 }
 ```
 
-Note that `Params` encodes much information from the blockchain direction, the `Context` in other terms. This is validated data and can be trusted, to compare any messages against. Refer to [the standard cosmwasm types](https://github.com/confio/cosmwasm/blob/master/src/types.rs#L3-L36) for any references you want to the environment. 
+Note that `Params` encodes much information from the blockchain direction, the `Context` in other terms [__I don't really understand this sentence__]. This is validated data and can be trusted to compare any messages against. Refer to [the standard cosmwasm types](https://github.com/confio/cosmwasm/blob/master/src/types.rs#L3-L36) for any references you want to add to the environment.
```
Comment (okwme) [inline docs/intro/editing-escrow-contract.md]:
```suggestion
Note that `Params` encodes a lot of information from the blockchain, essentially providing the `Context`. This is validated data and can be trusted to compare any messages against. Refer to [the standard cosmwasm types](https://github.com/confio/cosmwasm/blob/master/src/types.rs#L3-L36) for references to all the available types in the environment.
```

**Comment by okwme** [comment]
Something strange here: I made some changes to the contract that aren't in the original (they're in an open PR atm). However when I run the `cosmwasm-opt` command and run the checksum on the resulting file, it still matches the one in the tutorial. Mayyybe I only altered from the suggestions on the tests though, are those not included in the compile?

**Comment by ethanfrey** [comment]
Tests are indeed not included in the compile.
It never compiles test code in production releases, and actually very aggressively trims out dead code, so even non-test functions that can be proven never to be called by the `extern "C"` entrypoints will be removed. Even if used in helpers and called from other functions... unless they are actually called in some production code path, they will be removed by cosmwasm-opt. (and not effect hash).

Adding the steal code should change it. Then updating the theif address should also modify it

**Review by ethanfrey — APPROVED** [review]
Good cleanup. And good questions. I will merge, then address the questions. Maybe add a bit more to the "tests inside contracts.rs" part.

**Pull request merged**